test/distributed/test_functional_api.py

```diff
@@ -607,6 +607,36 @@ def allreduce(t, pg):
         allreduce(torch.randn(8, device=self.device), pg=dist.group.WORLD)
 
 
+    @with_comms
+    @requires_nccl()
+    @skip_if_lt_x_gpu(WORLD_SIZE)
+    def test_permute_tensor(self):
+
+        # rank0: [0., 1.], rank1: [2., 3.]
+        send_tensor = torch.arange(2, dtype=torch.float32, device="cuda") + 2 * self.rank
+        recvd_tensor = ft_c.permute_tensor(
+            send_tensor,
+            [
+                (0, 1),
+                (1, 0)
+            ],
+            group=dt.DeviceMesh("cuda", torch.arange(self.world_size))
+        )
+
+        # rank0: [2., 3.], rank1: [0., 1.]
+        expected = torch.arange(
+            2,
+            dtype=torch.float32,
+            device="cuda"
+        ) + 2 * ((self.rank - 1 + self.world_size) % self.world_size)
+        self.assertEqual(
```

Comment:
Let's also add a test about a sub-group case (or a device mesh that's created from a global mesh), and verify if it works.